src/config/mod.rs

```diff
@@ -393,25 +401,48 @@ mod tests {
             .build()
             .unwrap();
 
-        assert_eq!(config.initrd_path.unwrap().as_os_str(), "/tmp/initrd.img");
+        assert_eq!(
+            config.initrd_path.as_ref().unwrap().as_os_str(),
+            "/tmp/initrd.img"
+        );
         assert_eq!(
             config
                 .host_initrd_path
+                .as_ref()
```

Comment:
and this?

Reply:
because of this line, where config is borrowed
https://github.com/blockjoy/firec/pull/14/commits/cbd16a43e282c82dd4e2da131a145fc87b593192#diff-b0ba3e54b6f5ef1fa6aed8075e352a68fda66067fbafa8421042a2347801338aR443

Reply:
another way around is to call host_initrd_path() i suppose

Reply:
> because of this line, where config is borrowed
https://github.com/blockjoy/firec/commit/cbd16a43e282c82dd4e2da131a145fc87b593192#diff-b0ba3e54b6f5ef1fa6aed8075e352a68fda66067fbafa8421042a2347801338aR443

Right so just need to be moved into that commit.

Reply:
> another way around is to call host_initrd_path() i suppose

That'd be better I think.

Reply:
oh we don't have that func :)

Reply:
i mean in this commit, and in the following we remove those as_ref's

Reply:
> Right so just need to be moved into that commit.

it's related to changes in this commit

Reply:
how so, the type of those fields is not changing and neither is the string you're comparing with here.